src/ucp/core/ucp_ep.c

```diff
@@ -2771,7 +2771,7 @@ int ucp_ep_do_keepalive(ucp_ep_h ep)
 
     UCP_WORKER_THREAD_CS_CHECK_IS_BLOCKED(ep->worker);
 
-    if (ep->flags & UCP_EP_FLAG_FAILED) {
+    if ((ep->flags & UCP_EP_FLAG_FAILED) || (worker->keepalive.lane_map == 0)) {
```

Comment:
i think better do it after line 2780

Reply:
why there?

Reply:
seems it can be non-zero before this line, and become 0 after &= with ep_check_map

Reply:
done